src/main/java/com/aws/greengrass/mqttclient/spool/Spool.java

```diff
@@ -213,6 +238,67 @@ public long getCurrentSpoolerSize() {
     public SpoolerConfig getSpoolConfig() {
         return config;
     }
-}
 
+    /**
+     * Extract message ids from the persistenceSpool plugin's on disk database and insert the message \
+     * ids into queueOfMessageId, this function is only used in FileSystem storage mode.
+     * @param diskQueueOfIds list of messageIds to sync
+     * @param persistenceSpool instance of CloudMessageSpool
+     */
+    private void persistentQueueSync(List<Long> diskQueueOfIds, CloudMessageSpool persistenceSpool) {
+        if (!diskQueueOfIds.isEmpty()) {
+            long highestId = -1;
+            int i;
+            for (i = 0; i < diskQueueOfIds.size(); i += 1) {
+                //Check for queue space and remove if necessary
+                long currentId = diskQueueOfIds.get(i);
+                SpoolMessage message = persistenceSpool.getMessageById(currentId);
+                PublishRequest request = message.getRequest();
+                try {
+                    queueCapacityCheck(request);
+
+                    queueOfMessageId.putLast(currentId);
+                    if (currentId > highestId) {
+                        highestId = currentId;
+                    }
+                } catch (SpoolerStoreException e) {
+                    logger.atError().log(e);
+                    throw new RuntimeException(e);
```

Comment:
Can we directly throw `SpoolerStoreException` ? We usually avoid log _and_ throw: either log or throw.

Reply:
I will throw the SpoolerStoreException as a ServiceLoadException so that it will be handled by the setUpSpooler() method.